--- src/tdenetwork/kopete/libkopete/kautoconfig.cpp-orig	2016-11-30 16:39:03.450585866 +0200
+++ src/tdenetwork/kopete/libkopete/kautoconfig.cpp	2016-11-30 16:42:34.917998760 +0200
@@ -31,35 +31,13 @@
  * Macro function to warn developers when they are making calls
  * that can never return anything of value
  */ 
-#ifndef NDEBUG
-#include "kdebug.h"
-#define functionCallPreOrderCheck(functionName, returnValue) \
-  if(!d->retrievedSettings){ \
-      kdDebug(180) << "KAutoConfig::"functionName"() was called before " \
-      "KAutoConfig::retrieveSettings().  This should NEVER happen because " \
-      "it will do nothing.  Please Fix." << endl; \
-    return returnValue; \
-  }
-
-#define functionCallPostOrderCheck(functionName, returnValue) \
-  if(d->retrievedSettings){ \
-      kdDebug(180) << "KAutoConfig::"functionName"() was called after " \
-      "KAutoConfig::retrieveSettings().  This should NEVER happen because " \
-      "it will do nothing.  Please Fix." << endl; \
-    return returnValue; \
-  }
-#else
 #define functionCallPostOrderCheck(functionName, returnValue)
 #define functionCallPreOrderCheck(functionName, returnValue)
-#endif
 
 class KAutoConfig::KAutoConfigPrivate {
 
 public:
   KAutoConfigPrivate() : changed(false)
-#ifndef NDEBUG
-    , retrievedSettings(false)
-#endif
   { init(); }
 
   // Widgets to parse
@@ -73,11 +51,6 @@
   // Reset to false after saveSettings returns true.
   bool changed;
 
-#ifndef NDEBUG
-  // Many functions require this to be true to be of any value.
-  bool retrievedSettings;
-#endif
-
   // Known widgets that can be configured
   TQMap<TQWidget*, TQPtrList<TQWidget> > autoWidgets;
   // Default values for the widgets.
@@ -129,15 +102,6 @@
 }
 
 bool KAutoConfig::retrieveSettings(bool trackChanges){
-#ifndef NDEBUG
-  if(d->retrievedSettings){
-      kdDebug(180) << "This should not happen.  Function "
-       "KAutoConfig::retrieveSettings() was called more then once, returning "
-       "false.  Please fix." << endl;
-    return false;
-  }
-  d->retrievedSettings = true;
-#endif
   
   if(trackChanges){
     // QT
@@ -372,15 +336,6 @@
     }
 
     bool parseTheChildren = true;
-#ifndef NDEBUG
-    if(d->ignoreTheseWidgets[childWidget->className()] == 0 &&
-		    childWidget->name(0) == NULL){
-      // Without a name the widget is just skipped over.
-      kdDebug(180) << "KAutoConfig::retrieveSettings, widget with "
-        "NULL name.  className: " << childWidget->className() << endl;
-    }
-#endif
-    
     
     if( d->ignoreTheseWidgets[childWidget->className()] == 0 &&  
       childWidget->name(0) != NULL )
@@ -414,27 +369,7 @@
           connect(childWidget, changedMap[TQString::fromLatin1(childWidget->className())],
                   this, TQT_SIGNAL(widgetModified()));
         }
-#ifndef NDEBUG
-        else if(trackChanges &&
-          changedMap.find(TQString::fromLatin1(childWidget->className())) == changedMap.end())
-        {
-          // Without a signal kautoconfigdialog could incorectly 
-	  // enable/disable the buttons
-	  kdDebug(180) << "KAutoConfig::retrieveSettings, Unknown changed "
-           "signal for widget:" << childWidget->className() << endl;
-        }
-#endif
-
       }
-#ifndef NDEBUG
-      else
-      { 
-        // If kautoconfig doesn't know how to get/set the widget's value 
-	// nothing can be done to it and it is skipped. 
-	kdDebug(180) << "KAutoConfig::retrieveSettings, Unknown widget:" 
-          << childWidget->className() << endl;
-      }
-#endif
     }
     if(parseTheChildren)
     {
