--- a/dependencies/arts/mcop_mt/threads_posix.cc    2015-09-22 22:35:33.000000000 -0700
+++ b/dependencies/arts/mcop_mt/threads_posix.cc    2018-06-11 15:36:11.579041600 -0700
@@ -28,6 +28,7 @@
 #ifdef HAVE_LIBPTHREAD

 #include <gsl/gslconfig.h>
+#include <gsl/gslcommon.h>

 #include <sys/types.h>
 #include <stddef.h>
@@ -332,7 +333,7 @@
        }
 } initOnStartup;

-#include <gsl/gslcommon.h>
+//#include <gsl/gslcommon.h>
 /* -fast- locking for gsl on platforms with unix98 support */
 #if (GSL_HAVE_MUTEXATTR_SETTYPE > 0)
 static void pth_mutex_init (GslMutex *mutex)
