diff -Naur tdelibs-orig/CMakeLists.txt tdelibs/CMakeLists.txt
--- tdelibs-orig/CMakeLists.txt	2016-11-06 21:15:25.000000000 -0800
+++ tdelibs/CMakeLists.txt	2017-06-28 06:30:28.049251610 -0700
@@ -897,9 +897,9 @@
 ##### check for openssl #########################
 
 if( WITH_SSL )
-  pkg_search_module( SSL openssl )
+  pkg_search_module( SSL openssl<=1.0.21 )
   if( NOT SSL_FOUND )
-    message(FATAL_ERROR "\nSSL support are requested, but openssl is not found on your system" )
+    message(FATAL_ERROR "\nSSL support are requested, but openssl<=1.0.21 is not found on your system" )
   endif( NOT SSL_FOUND )
   set( KSSL_HAVE_SSL 1 )
   set( HAVE_SSL 1 )
diff -Naur tdelibs-orig/tdecert/CMakeLists.txt tdelibs/tdecert/CMakeLists.txt
--- tdelibs-orig/tdecert/CMakeLists.txt	2015-09-22 19:31:21.000000000 -0700
+++ tdelibs/tdecert/CMakeLists.txt	2017-06-28 06:29:12.715918239 -0700
@@ -22,10 +22,12 @@
   ${CMAKE_SOURCE_DIR}/tdeio/tdeio
   ${CMAKE_SOURCE_DIR}/tdeio/tdefile
   ${CMAKE_SOURCE_DIR}/tdeio/kssl
+  ${SSL_INCLUDE_DIRS}
 )
 
 link_directories(
   ${TQT_LIBRARY_DIRS}
+  ${SSL_LIBRARY_DIRS}
 )
 
 
diff -Naur tdelibs-orig/tdeio/kssl/CMakeLists.txt tdelibs/tdeio/kssl/CMakeLists.txt
--- tdelibs-orig/tdeio/kssl/CMakeLists.txt	2015-09-22 19:31:21.000000000 -0700
+++ tdelibs/tdeio/kssl/CMakeLists.txt	2017-06-28 06:21:09.835917933 -0700
@@ -22,8 +22,12 @@
   ${CMAKE_SOURCE_DIR}/tdecore/network
   ${CMAKE_SOURCE_DIR}/tdeui
   ${CMAKE_SOURCE_DIR}/tdewallet/client
+  ${SSL_INCLUDE_DIRS}
 )
 
+link_directories(
+  ${SSL_LIBRARY_DIRS}
+)
 
 ##### headers ###################################
 
